src/openssl/ssl/server.cr

```diff
@@ -72,6 +72,13 @@ class OpenSSL::SSL::Server
     end
   end
 
+  # This method allows to fetch the SSL SNI string if present
+  def hostname : String?
+    if host_name = LibSSL.ssl_get_servername(@ssl, LibSSL::TLSExt::NAMETYPE_host_name)
```

Comment:
Where is `@ssl` declared? And shouldn't hostname be a property of the SSL server socket? The point of this is that you have *one*  server wich can accepts connections with different hostnames.

Reply:
@straight-shoota -> https://github.com/crystal-lang/crystal/blob/530b9134582f9c97feea381c23d8e66037e72e8d/src/openssl/ssl/socket.cr#L82

Reply:
This is `OpenSSL::SSL::Server`, not `OpenSSL::SSL::Socket::Server`. Did you mix up the classes?

Reply:
These names *are* exceptionally confusing. That needs to improve with the new generic TLS API.

Reply:
Oh! damn yes I did, fixing now! damn...

Reply:
@straight-shoota fixed, sorry